brainscore_vision/nwb_integration/create_assembly.py

```diff
@@ -0,0 +1,209 @@
+        label_item      = nwb_file.electrodes['label'][i]
+        try: label_item = int(nwb_file.electrodes['label'][i].split('_')[0])
+        except: label_item = nwb_file.electrodes['label'][i]
+
+        location_match = re.search(r'\[(\d+).0, (\d+).0, (\d+).0\]', location_item)
+        if location_match:
+            data_dict['col'] = location_match.group(2)
+            data_dict['row'] = location_match.group(1)
+            data_dict['elec'] = location_match.group(3) 
+        else:
+            data_dict['col'] = None
+            data_dict['row'] = None
+            data_dict['elec'] = None
+
+        serialnumer = group_item.description.split('Serialnumber: ')[-1]
+        data_dict['arr'] = serialnumer
+
+        group_match = re.search(r"\['(\w+)', '(\w+)', '(\w+)'\]", group_item.location)
+        if group_match:
+            data_dict['hemisphere']  = group_match.group(1)
+            data_dict['region']  = group_match.group(2)
+            data_dict['subregion'] = group_match.group(3)
+        else:
+            data_dict['hemisphere'] = None
+            data_dict['region'] = None
+            data_dict['subregion'] = None
+        
+        data_dict['bank']  = bank_item.split('_')[-1]
+        data_dict['animal'] = subject
+        if (label_item) < 10:
+            neuroid_id = f"{bank_item.split('_')[-1]}-00{label_item}"
+            elec = f"00{label_item}"
+        else:
+            neuroid_id = f"{bank_item.split('_')[-1]}-0{label_item}"
+            elec = f"00{label_item}"
+        data_dict['neuroid_id']  = neuroid_id
+        data_dict['elec']  = elec
+
+        data_list.append(data_dict)
+    
+    neuroid_meta = pd.DataFrame(data_list)
+    return neuroid_meta
+
+def get_QC_neurids(nwb_file: NWBFile) -> Tuple[np.array, np.array]:
+    #-----------------------------------------------------------------------------------------------------------------------------
+    # This Method uses logical OR to find the common QC channels. (Closer to the BrainScore Method)
+    #-----------------------------------------------------------------------------------------------------------------------------
+    psth = nwb_file.scratch['PSTHs_ZScored_SessionMerged'][:]
+    common_QC_channels = np.logical_and.reduce(nwb_file.scratch['QualityApprovedChannelMasks'])
+    channel_masks_day = nwb_file.scratch['QualityApprovedChannelMasks'][:]
+    channel_mask_all_list = []
+    day = 0
+    for key in sorted(nwb_file.scratch.keys()):
+        if key.startswith('PSTHs_QualityApproved_20'):
+            nreps_per_day = nwb_file.scratch[key][:].shape[1]
+            for i in range(nreps_per_day):
+                channel_mask_all_list.append(channel_masks_day[day,:])
+            day += 1
+    channel_mask_all = np.array(channel_mask_all_list)
+
+    assert channel_mask_all.shape[0] == psth.shape[1]
+    filtered_neurids = np.any(channel_mask_all, axis=0)
+    return filtered_neurids, common_QC_channels
```

Comment:
I think my comment about documenting the shape of the outputs in the function string is probably going to apply to all of the functions, this is deep in the weeds of the NWB format